fdbserver/workloads/DataDistributionMetrics.actor.cpp

```diff
@@ -26,52 +26,110 @@
+	PerfIntCounter commits, errors;
 
 	DataDistributionMetricsWorkload(WorkloadContext const& wcx)
-	  : KVWorkload(wcx), transactionsCommitted(0), numShards(0), avgBytes(0) {
-		numTransactions = getOption(options, LiteralStringRef("numTransactions"), 100);
-		writesPerTransaction = getOption(options, LiteralStringRef("writesPerTransaction"), 1000);
+	  : KVWorkload(wcx), numShards(0), avgBytes(0), commits("Commits"), errors("Errors") {
+		testDuration = getOption(options, LiteralStringRef("testDuration"), 10.0);
+		keyPrefix = getOption(options, LiteralStringRef("keyPrefix"), LiteralStringRef("DDMetrics")).toString();
+		readPerTx = getOption(options, LiteralStringRef("readPerTransaction"), 1);
+		writePerTx = getOption(options, LiteralStringRef("writePerTransaction"), 5 * readPerTx);
+		ASSERT(nodeCount > 1);
 	}
 
 	static Value getRandomValue() {
 		return Standalone<StringRef>(format("Value/%08d", deterministicRandom()->randomInt(0, 10e6)));
 	}
 
-	ACTOR static Future<Void> _start(Database cx, DataDistributionMetricsWorkload* self) {
-		state int tNum;
-		for (tNum = 0; tNum < self->numTransactions; ++tNum) {
-			loop {
-				state ReadYourWritesTransaction tr(cx);
-				try {
-					state int i;
-					for (i = 0; i < self->writesPerTransaction; ++i) {
-						tr.set(StringRef(format("Key/%08d", tNum * self->writesPerTransaction + i)), getRandomValue());
+	Key keyForIndex(int n) { return doubleToTestKey((double)n / nodeCount, keyPrefix); }
+
+	ACTOR static Future<Void> ddRWClient(Database cx, DataDistributionMetricsWorkload* self) {
+		loop {
+			state ReadYourWritesTransaction tr(cx);
+			state int i;
+			try {
+				for (i = 0; i < self->readPerTx; ++i)
+					wait(success(tr.get(self->keyForIndex(deterministicRandom()->randomInt(0, self->nodeCount))))); // read
+				for (i = 0; i < self->writePerTx; ++i)
+					tr.set(self->keyForIndex(deterministicRandom()->randomInt(0, self->nodeCount)), getRandomValue()); // write
+				wait(tr.commit());
+				++self->commits;
+			} catch (Error& e) {
+				wait(tr.onError(e));
+			}
+			tr.reset();
+		}
+	}
+
+	ACTOR Future<Void> resultConsistencyCheckClient(Database cx, DataDistributionMetricsWorkload* self) {
+		state Reference<ReadYourWritesTransaction> tr =
+		    Reference<ReadYourWritesTransaction>(new ReadYourWritesTransaction(cx));
+		loop {
+			try {
+				int startIndex = deterministicRandom()->randomInt(0, self->nodeCount - 1);
+				int endIndex = deterministicRandom()->randomInt(startIndex + 1, self->nodeCount);
+				state Key startKey = self->keyForIndex(startIndex);
+				state Key endKey = self->keyForIndex(endIndex);
+				// lastLessOrEqual
+				state KeySelector begin = KeySelectorRef(startKey.withPrefix(ddStatsRange.begin, startKey.arena()), true, 0);
+				state KeySelector end = KeySelectorRef(endKey.withPrefix(ddStatsRange.begin, endKey.arena()), false, 2);
+				Standalone<RangeResultRef> result = wait(tr->getRange(begin, end, GetRangeLimits(CLIENT_KNOBS->SHARD_COUNT_LIMIT)));
```

Comment:
This is a cool way to test this weakened getRange contract where reads aren't necessarily repeatable. Nice job!